src/emoji/EmojiPickerScreen.js

```diff
@@ -0,0 +1,86 @@
+/* @flow */
+import { connect } from 'react-redux';
+
+import React, { Component } from 'react';
+import { View, Text, FlatList } from 'react-native';
+import type { NavigationScreenProp } from 'react-navigation';
+
+import { emojiReactionAdd } from '../api';
+import codePointMap from './codePointMap';
+import { Screen, Input } from '../common';
+import EmojiRow from './EmojiRow';
+import getFilteredEmojiList from './getFilteredEmojiList';
+import type { GlobalState, RealmEmojiState, Auth, Dispatch } from '../types';
+import { getActiveRealmEmoji, getAuth } from '../selectors';
+import { navigateBack } from '../nav/navActions';
+
+type Props = {
+  realmEmoji: RealmEmojiState,
+  auth: Auth,
+  dispatch: Dispatch,
+  navigation: NavigationScreenProp<*> & {
+    state: {
+      params: {
+        messageId: number,
+      },
+    },
+  },
+};
+
+type State = {
+  filter: string,
+};
+
+class EmojiPickerScreen extends Component<Props, State> {
+  props: Props;
+  state: State;
+
+  state = {
+    filter: '',
+  };
+
+  handleInputChange = text => {
+    this.setState({
+      filter: text.toLowerCase(),
+    });
+  };
+
+  addReaction = item => {
+    const { auth, dispatch } = this.props;
+    const { messageId } = this.props.navigation.state.params;
+    emojiReactionAdd(auth, messageId, 'unicode_emoji', codePointMap[item], item);
+    dispatch(navigateBack());
+  };
+
+  render() {
+    const { realmEmoji } = this.props;
+    const { filter } = this.state;
+
+    const emojis = getFilteredEmojiList(filter, realmEmoji);
+
+    return (
+      <Screen search searchBarOnChange={this.handleInputChange}>
+        <FlatList
+          keyboardShouldPersistTaps="always"
+          initialNumToRender={20}
+          data={emojis}
+          keyExtractor={item => item}
```

Comment:
That is probably not correct.
What you need is a value from this `item` object that uniquely identifies it.
The emoji `name` prop is likely good for that.

Reply:
`item` in this case is the emoji name, it's not an object.